react/week2/todo-list/src/App.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from "react";
+import "./App.css";
+
+const todos = [
+  {
+    id: 1,
+    description: "Get out of bed",
+  },
+  {
+    id: 2,
+    description: "Brush teeth",
+  },
+  {
+    id: 3,
+    description: "Eat breakfast",
+  },
+  {
+    id: 4,
+    description: "go school",
+  },
+];
+
+const TodoTimer = (props) => {
+  const [count, setCount] = useState(0);
+  useEffect(() => {
+    setTimeout(() => setCount(count + 1), 1000);
+  });
+
+  return <p>You have used this app for {count} seconds</p>;
+};
+
+const NotItem = (props) => {
+  return <span>{props.text} </span>;
+};
+
+const TodoRow = (props) => {
+  const [checked, setChecked] = useState(false);
+  const itemChecked = checked ? "label" : "";
+  return (
+    <li>
+      <label id={props.id} className={itemChecked}>
+        {props.textDescription}
+      </label>
+      <input
+        id={props.id}
+        type="checkbox"
+        onChange={() => setChecked(!checked)}
+      ></input>
+      <button
+        style={{ fontSize: 15 }}
+        onClick={() => props.handleDelete(props.id)}
+      >
+        Delete
+      </button>
+    </li>
+  );
+};
+
+const TodoList = (props) => {
+  if (props.todorows.length === 0) {
+    return <NotItem text="No items" />;
+  }
+
+  return (
+    <ul>
+      {props.todorows.map((x, index) => {
+        return (
+          <TodoRow
+            key={index}
+            textDescription={x.description}
```

Comment:
The value of the `key` attribute should have a unique and stable value. The value of an item's index changes when list is updated. You can add the item's `id`

```suggestion
          <TodoRow
            key={x.id}
            textDescription={x.description}
```

Reply:
Thanks Abdualrhman for great comments 👍